File: packages/ash-php/src/Core/AshErrorCode.php
<?php

declare(strict_types=1);

namespace Ash\Core;

/**
 * Error codes returned by ASH verification.
 */
enum AshErrorCode: string
{
    case InvalidContext = 'ASH_INVALID_CONTEXT';
    case ContextExpired = 'ASH_CONTEXT_EXPIRED';
    case ReplayDetected = 'ASH_REPLAY_DETECTED';
    case IntegrityFailed = 'ASH_INTEGRITY_FAILED';
    case EndpointMismatch = 'ASH_ENDPOINT_MISMATCH';
    case ModeViolation = 'ASH_MODE_VIOLATION';
    case UnsupportedContentType = 'ASH_UNSUPPORTED_CONTENT_TYPE';
    case MalformedRequest = 'ASH_MALFORMED_REQUEST';
    case CanonicalizationFailed = 'ASH_CANONICALIZATION_FAILED';
}

File: packages/ash-dotnet/src/Ash.Core/AshErrorCode.cs
// ASH was developed by 3maem Co. | 12/31/2025

namespace Ash.Core;

/// <summary>
/// Error codes returned by ASH verification.
/// </summary>
public static class AshErrorCode
{
    /// <summary>Context not found or invalid.</summary>
    public const string InvalidContext = "ASH_INVALID_CONTEXT";

    /// <summary>Context has expired.</summary>
    public const string ContextExpired = "ASH_CONTEXT_EXPIRED";

    /// <summary>Request replay detected - context already consumed.</summary>
    public const string ReplayDetected = "ASH_REPLAY_DETECTED";

    /// <summary>Proof verification failed - payload may have been tampered.</summary>
    public const string IntegrityFailed = "ASH_INTEGRITY_FAILED";

    /// <summary>Context binding does not match requested endpoint.</summary>
    public const string EndpointMismatch = "ASH_ENDPOINT_MISMATCH";

    /// <summary>Mode violation detected.</summary>
    public const string ModeViolation = "ASH_MODE_VIOLATION";

    /// <summary>Content type not supported by ASH protocol.</summary>
    public const string UnsupportedContentType = "ASH_UNSUPPORTED_CONTENT_TYPE";

    /// <summary>Malformed request.</summary>
    public const string MalformedRequest = "ASH_MALFORMED_REQUEST";

    /// <summary>Failed to canonicalize payload.</summary>
    public const string CanonicalizationFailed = "ASH_CANONICALIZATION_FAILED";
}
